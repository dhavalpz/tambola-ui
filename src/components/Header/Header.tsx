import Button from "@mui/material/Button";
import React from "react";
import { useContext } from "react";
import { ThemeContext } from "../ThemeSetter";

const Header: React.FC = () => {
  const setThemeName = useContext(ThemeContext);

  return (
    <>
      <Button
        variant="contained"
        color="primary"
        onClick={() => setThemeName("lightTheme")}
      >
        Light Theme
      </Button>
      <Button
        variant="contained"
        color="primary"
        onClick={() => setThemeName("darkTheme")}
      >
        Dark Theme
      </Button>
    </>
  );
};
export { Header };
