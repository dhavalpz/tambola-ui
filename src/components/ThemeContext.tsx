import React from "react";
import {
  createTheme,
  ThemeProvider as MuiThemeProvider,
} from "@mui/material/styles";
import { themeCreator } from "./Theme/BaseTheme";

const ThemeContext = React.createContext((_: string) => {});

const APPTHEME = "appTheme";

const ThemeProvider: React.FC = (props) => {
  const currentThemeName = localStorage.getItem(APPTHEME) || "lightTheme";

  const [themeName, setThemeName] = React.useState(currentThemeName);

  const theme = React.useMemo(
    () => createTheme(themeCreator(themeName)),
    [themeName]
  );

  const setTheme = (themeName: string): void => {
    localStorage.setItem(APPTHEME, themeName);
    setThemeName(themeName);
  };

  return (
    <ThemeContext.Provider value={setTheme}>
      <MuiThemeProvider theme={theme}>{props.children}</MuiThemeProvider>
    </ThemeContext.Provider>
  );
};

export { ThemeProvider, ThemeContext };
