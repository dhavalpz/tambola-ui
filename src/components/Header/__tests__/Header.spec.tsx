import React from "react";
import { Header } from "../Header";
import { fireEvent, render, screen } from "@testing-library/react";
import { ThemeContext } from "../../ThemeContext";

describe("Header", () => {
  it("Header Match snapshot test", () => {
    const tree = render(<Header />);

    expect(tree).toMatchSnapshot();
  });

  it("should get light theme ", () => {
    const setThemeName = jest.fn();

    render(
      <ThemeContext.Provider value={setThemeName}>
        <Header />
      </ThemeContext.Provider>
    );

    fireEvent.click(screen.getByTestId("lightButton"));

    expect(setThemeName).toHaveBeenCalledTimes(1);
    expect(setThemeName).toHaveBeenCalledWith("lightTheme");
  });

  it("should get dark theme ", () => {
    const setThemeName = jest.fn();

    render(
      <ThemeContext.Provider value={setThemeName}>
        <Header />
      </ThemeContext.Provider>
    );

    fireEvent.click(screen.getByTestId("darkButton"));

    expect(setThemeName).toHaveBeenCalledTimes(1);
    expect(setThemeName).toHaveBeenCalledWith("darkTheme");
  });
});
