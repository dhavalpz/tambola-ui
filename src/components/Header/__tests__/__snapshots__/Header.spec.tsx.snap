// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header Match snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-1n0nzlo-MuiTypography-root"
            >
              TAMBOLA
            </h4>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButtonBase-root css-11jqx3f-MuiButtonBase-root-MuiButton-root"
              data-testid="lightButton"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Brightness7OutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zm-2 5.79V18h-3.52L12 20.48 9.52 18H6v-3.52L3.52 12 6 9.52V6h3.52L12 3.52 14.48 6H18v3.52L20.48 12 18 14.48zM12 6.5c-3.03 0-5.5 2.47-5.5 5.5s2.47 5.5 5.5 5.5 5.5-2.47 5.5-5.5-2.47-5.5-5.5-5.5zm0 9c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="2"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButtonBase-root css-11jqx3f-MuiButtonBase-root-MuiButton-root"
              data-testid="darkButton"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Brightness2OutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 4c4.41 0 8 3.59 8 8s-3.59 8-8 8c-.34 0-.68-.02-1.01-.07C10.9 17.77 12 14.95 12 12s-1.1-5.77-3.01-7.93C9.32 4.02 9.66 4 10 4m0-2c-1.82 0-3.53.5-5 1.35C7.99 5.08 10 8.3 10 12s-2.01 6.92-5 8.65C6.47 21.5 8.18 22 10 22c5.52 0 10-4.48 10-10S15.52 2 10 2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-1n0nzlo-MuiTypography-root"
          >
            TAMBOLA
          </h4>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButtonBase-root css-11jqx3f-MuiButtonBase-root-MuiButton-root"
            data-testid="lightButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="Brightness7OutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zm-2 5.79V18h-3.52L12 20.48 9.52 18H6v-3.52L3.52 12 6 9.52V6h3.52L12 3.52 14.48 6H18v3.52L20.48 12 18 14.48zM12 6.5c-3.03 0-5.5 2.47-5.5 5.5s2.47 5.5 5.5 5.5 5.5-2.47 5.5-5.5-2.47-5.5-5.5-5.5zm0 9c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
              />
              <circle
                cx="12"
                cy="12"
                r="2"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-outlined MuiButton-outlinedInherit MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorInherit MuiButtonBase-root css-11jqx3f-MuiButtonBase-root-MuiButton-root"
            data-testid="darkButton"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="Brightness2OutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 4c4.41 0 8 3.59 8 8s-3.59 8-8 8c-.34 0-.68-.02-1.01-.07C10.9 17.77 12 14.95 12 12s-1.1-5.77-3.01-7.93C9.32 4.02 9.66 4 10 4m0-2c-1.82 0-3.53.5-5 1.35C7.99 5.08 10 8.3 10 12s-2.01 6.92-5 8.65C6.47 21.5 8.18 22 10 22c5.52 0 10-4.48 10-10S15.52 2 10 2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
